src/main/scala/pt/tecnico/dsi/openstack/designate/services/ZoneTransferRequests.scala

```diff
@@ -0,0 +1,42 @@
+package pt.tecnico.dsi.openstack.designate.services
+
+import cats.effect.Sync
+import cats.syntax.flatMap._
+import fs2.Stream
+import io.circe.Encoder
+import org.http4s.client.Client
+import org.http4s.{Header, Query, Uri}
+import pt.tecnico.dsi.openstack.common.models.WithId
+import pt.tecnico.dsi.openstack.common.services.Service
+import pt.tecnico.dsi.openstack.designate.models.{Status, ZoneTransferRequest}
+
+// This class does not extend CrudService because `create` receives an extra zoneId parameter.
+
+final class ZoneTransferRequests[F[_]: Sync: Client](baseUri: Uri, authToken: Header, createUri: String => Uri) extends Service[F](authToken) {
+  val uri: Uri = baseUri / "transfer_requests"
+
+  def list(status: Status, extraHeaders: Header*): Stream[F, WithId[ZoneTransferRequest]] =
+    list(Query.fromPairs("status" -> status.toString), extraHeaders:_*)
+
+  def list(extraHeaders: Header*): Stream[F, WithId[ZoneTransferRequest]] = list(Query.empty, extraHeaders:_*)
+
+  def list(query: Query, extraHeaders: Header*): Stream[F, WithId[ZoneTransferRequest]] =
+    super.list[WithId[ZoneTransferRequest]]("transfer_requests", uri, query, extraHeaders:_*)
+
+  def create(zoneId: String, value: ZoneTransferRequest.Create, extraHeaders: Header*): F[WithId[ZoneTransferRequest]] =
+    super.postHandleConflict(wrappedAt = None, value, createUri(zoneId) / "transfer_requests", extraHeaders:_*) {
+      list(Query.empty, extraHeaders:_*).filter(_.zoneId == zoneId).head.compile.lastOrError.flatMap { existing =>
```

Comment:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)